ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/page.tsx

```diff
@@ -6,13 +8,23 @@ import { Button } from '@inngest/components/Button';
 import AppLink from '@/components/AppLink';
 import AppNavigation from '@/components/Navigation/AppNavigation';
 import Toaster from '@/components/Toaster';
-import getAllEnvironments from '@/queries/server-only/getAllEnvironments';
+import { useEnvironments } from '@/queries';
 import { EnvironmentType, LEGACY_TEST_MODE_NAME } from '@/utils/environments';
 import { EnvironmentArchiveButton } from './EnvironmentArchiveButton';
 import EnvironmentListTable from './EnvironmentListTable';
 
-export default async function Envs() {
-  const environments = await getAllEnvironments();
+export default function Envs() {
+  const [{ data: environments, fetching, error }] = useEnvironments();
+  if (error) {
+    throw error;
+  }
+  if (fetching) {
+    return null;
+  }
+  if (!environments) {
+    // Unreachable
+    throw new Error('No data');
```

Comment:
I wonder if we should have a clearer error here than just no-data. I understand that it says that it's unreachable, but we still should write a good error message in the case that our API or `useEnvironments` has bug. Maybe something like `Unable to load environments. Please try again or contact support if this continues`